### PR TITLE
feat(cloudweb): display default runtime in general informations tab

### PR DESCRIPTION
**Pull request by euhmeuh**

### CODE REVIEW

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+            this.HostingRuntimes = HostingRuntimes;
         }
 
         $onInit () {
             this.$scope.$on("hosting.ssl.reload", () => this.retrievingSSLCertificate());
 
+            this.defaultRuntime = null;
+            this.HostingRuntimes.getDefault(this.$stateParams.productId)
+                .then((runtime) => {
+                    console.log(runtime);
```
Comment (cbourgois) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: -- `console.log`

Comment (cbourgois) [reply client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: no catch? what's happen if there is no default runtime (oui-spinner) ?

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 2]
```diff
@@ -363,6 +363,37 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
                                         <span data-ng-bind="hosting.filer"></span>
                                     </div>
                                 </li>
+                                <li class="oui-tile__item" data-ng-if="hosting.isCloudWeb">
+                                    <div class="oui-tile__definition">
+                                        <div class="d-flex w-100">
+                                            <div>
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: -- `div` ?

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 3]
```diff
@@ -363,6 +363,37 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                                <strong class="d-block"
+                                                        data-i18n-static="hosting_dashboard_service_default_runtime"></strong>
+                                                <span data-ng-bind="ctrl.defaultRuntime.name"
+                                                      data-ng-if="ctrl.defaultRuntime"></span>
+                                                <div class="d-inline ml-2" data-ng-if="!ctrl.defaultRuntime">
+                                                    <oui-spinner data-size="s"></oui-spinner>
+                                                </div>
+                                            </div>
+                                            <div class="mb-auto ml-auto">
+                                                <button class="oui-button oui-button_dropdown mb-0" type="button" aria-haspopup="true" aria-expanded="false"
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Didn't know about `aria-haspopup`:+1:

Comment (FredericEspiau) [reply client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: `aria-expanded`'s value should change wether the menu is expanded or not, doesn't it ?

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 4]
```diff
@@ -363,6 +363,37 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                                        data-toggle="dropdown">
+                                                    <span data-i18n-static="common_actions"></span>
+                                                    <span class="oui-icon oui-icon-chevron-down" aria-hidden="true"></span>
+                                                </button>
+                                                <ul class="dropdown-menu dropdown-menu-right m-0 p-0">
+                                                    <li>
+                                                        <i class="oui-icon oui-icon-power_line" aria-hidden="true"></i>
+                                                        <button class="btn btn-link" type="button" name="configureRuntimes"
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Too bad, you've used a `span` right above 😄

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 5]
```diff
@@ -363,6 +363,37 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                                                data-ng-click="ctrlHostingTabs.setSelectedTab('RUNTIMES')">
+                                                            <span data-i18n-static="common_configure"></span>
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: `data-translate` in the `button`

File: client/app/hosting/runtimes/hosting-runtimes.service.js [hunk 6]
```diff
@@ -58,6 +58,16 @@ angular
             });
         }
 
+        /**
+         * Get default runtime
+         * @param serviceName
+         */
+        getDefault (serviceName) {
+            return this.list(serviceName)
+                .then((runtimes) => this.$q.all(_(runtimes).map((runtimeId) => this.get(serviceName, runtimeId)).value()))
```
Comment (FredericEspiau) [inline client/app/hosting/runtimes/hosting-runtimes.service.js]: Alright, chains ❤️  !!!

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js [hunk 7]
```diff
@@ -1,18 +1,28 @@
...
             this.Alerter = Alerter;
             this.hostingSSLCertificate = hostingSSLCertificate;
             this.translator = translator;
+            this.HostingRuntimes = HostingRuntimes;
         }
 
         $onInit () {
             this.$scope.$on("hosting.ssl.reload", () => this.retrievingSSLCertificate());
 
+            this.defaultRuntime = null;
```
Comment (cbourgois) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: You should define `this.loading.defaultRuntime` to `true`, and change it to `false` in `finally`.

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 8]
```diff
@@ -363,6 +363,37 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
                                         <span data-ng-bind="hosting.filer"></span>
                                     </div>
                                 </li>
+                                <li class="oui-tile__item" data-ng-if="hosting.isCloudWeb">
+                                    <div class="oui-tile__definition">
+                                        <div class="d-flex w-100">
+                                            <div>
+                                                <strong class="d-block"
+                                                        data-i18n-static="hosting_dashboard_service_default_runtime"></strong>
+                                                <span data-ng-bind="ctrl.defaultRuntime.name"
```
Comment (cbourgois) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: `data-ng-if="!ctrl.loading.defaultRuntime"`

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js [hunk 9]
```diff
@@ -1,18 +1,32 @@
...
 
+            this.loading = {
+                defaultRuntime: true
+            };
+
+            this.defaultRuntime = null;
+            this.HostingRuntimes.getDefault(this.$stateParams.productId)
+                .then((runtime) => {
+                    this.defaultRuntime = runtime;
+                })
```
Comment (marie-j) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: No catching ?

Comment (euhmeuh) [reply client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: `finally` does the trick

Comment (antleblanc) [reply client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: Act more like displaying a message to the user that something went wrong.

Comment (euhmeuh) [reply client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js]: It actually does. Following @cbourgois's advice, another span shows'up in red when something goes wrong.

**Pull request merged**